tfx/components/evaluator/executor.py

```diff
@@ -176,52 +180,60 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
+      raise ValueError('example_splits in execution properties needs to be a '
+                       'list. Got %s instead.' % type(example_splits))
+
     absl.logging.info('Evaluating model.')
     with self._make_beam_pipeline() as pipeline:
+      examples_list = []
+      tensor_adapter_config = None
       # pylint: disable=expression-not-assigned
-      if _USE_TFXIO:
-        tensor_adapter_config = None
-        if tfma.is_batched_input(eval_shared_model, eval_config):
+      if _USE_TFXIO and tfma.is_batched_input(eval_shared_model, eval_config):
+        for split in example_splits:
+          file_pattern = io_utils.all_files_pattern(
+              artifact_utils.get_split_uri(input_dict[constants.EXAMPLES_KEY],
+                                           split))
           tfxio = tf_example_record.TFExampleRecord(
               file_pattern=file_pattern,
               schema=schema,
               raw_record_column_name=tfma_constants.ARROW_INPUT_COLUMN)
-          if schema is not None:
-            tensor_adapter_config = tensor_adapter.TensorAdapterConfig(
-                arrow_schema=tfxio.ArrowSchema(),
-                tensor_representations=tfxio.TensorRepresentations())
-          data = pipeline | 'ReadFromTFRecordToArrow' >> tfxio.BeamSource()
-        else:
-          data = pipeline | 'ReadFromTFRecord' >> beam.io.ReadFromTFRecord(
-              file_pattern=file_pattern)
-        (data
-         | 'ExtractEvaluateAndWriteResults' >>
-         tfma.ExtractEvaluateAndWriteResults(
-             eval_shared_model=models[0] if len(models) == 1 else models,
-             eval_config=eval_config,
-             output_path=output_uri,
-             slice_spec=slice_spec,
-             tensor_adapter_config=tensor_adapter_config))
+          data = (pipeline
+                  | 'ReadFromTFRecordToArrow[%s]' % split >>
+                  tfxio.BeamSource())
+          examples_list.append(data)
+        if schema is not None:
+          tensor_adapter_config = tensor_adapter.TensorAdapterConfig(
+              arrow_schema=tfxio.ArrowSchema(),
+              tensor_representations=tfxio.TensorRepresentations())
       else:
-        data = pipeline | 'ReadFromTFRecord' >> beam.io.ReadFromTFRecord(
-            file_pattern=file_pattern)
-        (data
-         | 'ExtractEvaluateAndWriteResults' >>
-         tfma.ExtractEvaluateAndWriteResults(
-             eval_shared_model=models[0] if len(models) == 1 else models,
-             eval_config=eval_config,
-             output_path=output_uri,
-             slice_spec=slice_spec))
+        for split in example_splits:
+          file_pattern = io_utils.all_files_pattern(
+              artifact_utils.get_split_uri(input_dict[constants.EXAMPLES_KEY],
+                                           split))
+          data = (pipeline
+                  | 'ReadFromTFRecord[%s]' % split >>
+                  beam.io.ReadFromTFRecord(file_pattern=file_pattern))
```

Comment:
check this API [ReadAllFromTFRecord](https://github.com/apache/beam/blob/b91560cc354da471e3de502aad78dd059997a3d0/sdks/python/apache_beam/io/tfrecordio.py#L206), we might be able to use this

Reply:
let's try this ReadAllFromTFRecord (check [here](https://github.com/apache/beam/search?q=ReadAllFromTFRecord&unscoped_q=ReadAllFromTFRecord) for the usage) for the else branch

line 234 need to be added to the if branch when we use  ReadAllFromTFRecord